apps/ledger-live-mobile/src/hooks/newsfeed/useCryptopanicPosts.ts

```diff
@@ -0,0 +1,101 @@
+import { useCallback, useEffect, useMemo, useState } from "react";
+import useFeature from "@ledgerhq/live-common/featureFlags/useFeature";
+import {
+  cryptopanicAvailableRegions,
+  CryptopanicAvailableRegionsType,
+  CryptopanicGetParams,
+  CryptopanicNewsWithMetadata,
+  getPosts,
+} from "./cryptopanicApi";
+import { useLocale } from "../../context/Locale";
+
+type LoadingStateType = null | "initial" | "refreshing" | "loadingMore";
+
+export function useCryptopanicPosts(
```

Comment:
several remarks about this hook:
we should avoid big "do-it-all" hooks like this for the sake of performance (especially that this one is used in a component that renders a list of items)
- I think the state should be kept in one or multiple Context providers at the top level of the app so that everything doesn't have to be re-initialised every time the hook is mounted (which will happen every time the user navigates to that page)
- then to avoid expensive re-renders of the list you might want to split it in several contexts:
  - one that just provides the list of posts that will be consumed and the bare minimum
  - one that provides loading state
  - another that provides access to the API (refresh, loadMore)
- then you split the big screen component so that each part (header, list of posts, footer) can use one of these contexts without being re-rendered uselessly and expensively every time the loading state changes)

... but yeah then you start see that using a Context for keeping frequently updating state like this is not ideal because (even if it's better than just a hook), so using redux here would make sense so that every little part of the screen can consume exactly what it needs in the store without being re-rendered too frequently